Shopify/ruby-lsp
# Add semantic token for constants

### Motivation

Inconsistent highlighting in constants: 

<img width="302" alt="Screen Shot 2022-06-08 at 8 39 31 PM" src="https://user-images.githubusercontent.com/25471753/172740317-10473c2c-7372-4535-b205-d1734498c5c8.png">

### Implementation

Add new semantic token type, `namespace`, and add them as a token for constants. 

Irrelevant to the primary issue but I also rearranged some of the visit methods in alphabetical order.  

### Automated Tests

- Fix the existing tests
- Add a new test file `const.exp`
- Fix the integration test -- the code contains a constant, so the assertion needed to be updated

### Manual Tests

1. Checkout this branch
2. Open `test/fixtures/const.rb`
3. See consistent highlighting: 

<img width="337" alt="Screen Shot 2022-06-08 at 8 44 15 PM" src="https://user-images.githubusercontent.com/25471753/172740691-2f0a3e55-770a-482a-8837-05a7e97c6b6c.png">
